### PR TITLE
feat: add gossip validator

### LINKED ISSUE #164: Libp2p: validate messages received from peers
This can be done with [`(*PubSub).RegisterTopicValidator`](https://pkg.go.dev/github.com/libp2p/go-libp2p-pubsub#PubSub.RegisterTopicValidator), which should receive a [`ValidatorEx`](https://pkg.go.dev/github.com/libp2p/go-libp2p-pubsub#ValidatorEx) so we can differentiate between _REJECT_ and _IGNORE_ actions.

### PR DESCRIPTION
**Pull request by MegaRedHand**
Depends on #324
Closes #164

### CODE REVIEW

File: lib/lambda_ethereum_consensus/p2p/subscriber.ex [hunk 1]
```diff
@@ -40,9 +40,11 @@ defmodule LambdaEthereumConsensus.P2P.Subscriber do
 
   @impl true
   def handle_info(
-        {:gossipsub, {topic, message}},
+        {:gossipsub, {topic, msg_id, message}},
         %{topic: topic, queue: queue, demand: demand} = state
       ) do
+    # TODO: validate
+    Libp2pPort.validate_message(msg_id, :accept)
```
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/p2p/subscriber.ex:47]: is it specified what a valid message means? For example, just that we can decode it or the contents as well?

What are the consequences of sending a reject or ignore? I guess that it is not going to be gossiped to other peers? Will the peer that sent this bad message be penalized somehow?

**Review by mpaulucci — APPROVED** [review]

File: lib/lambda_ethereum_consensus/p2p/subscriber.ex (hunk 1 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/subscriber.ex:47]: Oops. I forgot to document the function 😋. Will do that

Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/subscriber.ex:47]: Done: 449049e9a854ea13ca64ba2b22ce060377bc6063

Comment (mpaulucci) [reply lib/lambda_ethereum_consensus/p2p/subscriber.ex:47]: thanks!

**Pull request merged**